File: js/studentTopNotes.js
// Problem# 187
// Create a function that takes an array of students and returns an array of their top notes.
// If the student does not have notes then let's assume their top note is equal to 0.
// https://edabit.com/challenge/b7dXbWEhbr3bXCk7i

const getStudentTopNotes = (students) =>
  students.map(({ notes }) => (notes.length ? Math.max(...notes) : 0));

module.exports = getStudentTopNotes;

File: python/student_top_notes.py
"""
Get Student Top Notes

Create a function that takes a list of student dictionaries and returns a list of their top notes. If the student does not have notes then let's assume their top note is equal to 0.

https://edabit.com/challenge/jt8XTr75xxowgi6KM
"""
from typing import Dict, List


def get_student_top_notes(students: List[Dict]) -> List[int]:
    return [0 if not student['notes'] else max(student['notes']) for student in students]
